File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans=0,mini=prices[0];
        for(int i=1;i<prices.size();i++){
            int sp=prices[i];
            int prof=sp-mini;
            mini=min(mini,sp);
            ans=max(ans,prof);
        }
        return ans;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        profit,mini=0,prices[0]
        for i in range(1,len(prices)):
            price=prices[i]
            cur_profit=price-mini
            profit=max(profit,cur_profit)
            mini=min(mini,price)

        return profit
